src/logger/src/logger.rs

```diff
@@ -480,191 +432,262 @@ impl fmt::Display for LoggerError {
+            .is_ok());
+        validate_log(Box::new(&mut reader), &format!("{}\n", TEST_APP_HEADER));
+        // Check that the logs are written to the configured writer.
+        logger.mock_log(Level::Info, "info");
+        validate_log(
+            Box::new(&mut reader),
+            "[TEST-INSTANCE-ID:INFO:logger.rs:0] info\n",
         );
-        assert_eq!(
-            format!("{}", LoggerError::AlreadyInitialized),
-            "Reinitialization of logger not allowed."
+
+        // Assert that initialization works only once.
+        let (writer_2, mut reader_2) = log_channel();
+        assert!(logger
+            .init(TEST_APP_HEADER.to_string(), Box::new(writer_2))
+            .is_err());
+        // Check that the logs are written only to the first writer.
+        logger.mock_log(Level::Info, "info");
+        validate_log(
+            Box::new(&mut reader),
+            "[TEST-INSTANCE-ID:INFO:logger.rs:0] info\n",
         );
+        validate_log(Box::new(&mut reader_2), "");
+    }
 
-        assert_eq!(
-            format!("{}", LoggerError::IsInitializing),
-            "The logger is initializing. Can't perform the requested action right now."
+    #[test]
+    fn test_create_prefix() {
+        let logger = Logger::mock_new();
+        let mut reader = logger.mock_init();
+
+        // test with empty instance id
+        logger.set_instance_id("".to_string());
+
+        // check that the prefix is empty when all flags are false
+        logger
+            .set_include_level(false)
+            .set_include_origin(false, true);
+        logger.mock_log(Level::Info, "msg");
+        validate_log(Box::new(&mut reader), "[] msg\n");
+
+        // check that the prefix is correctly shown when all flags are true
+        logger
+            .set_include_level(true)
+            .set_include_origin(true, true);
+        logger.mock_log(Level::Info, "msg");
+        validate_log(Box::new(&mut reader), "[INFO:logger.rs:0] msg\n");
+
+        // check show_line_numbers=false
+        logger
+            .set_include_level(true)
+            .set_include_origin(true, false);
+        logger.mock_log(Level::Debug, "msg");
+        validate_log(Box::new(&mut reader), "[DEBUG:logger.rs] msg\n");
+
+        // check show_file_path=false
+        logger
+            .set_include_level(true)
+            .set_include_origin(false, true);
+        logger.mock_log(Level::Error, "msg");
+        validate_log(Box::new(&mut reader), "[ERROR] msg\n");
+
+        // check show_level=false
+        logger
+            .set_include_level(false)
+            .set_include_origin(true, true);
+        logger.mock_log(Level::Info, "msg");
+        validate_log(Box::new(&mut reader), "[logger.rs:0] msg\n");
+
+        // test with a mock instance id
+        logger.set_instance_id(TEST_INSTANCE_ID.to_string());
+
+        // check that the prefix is empty when all flags are false
```

Comment:
this comment is inaccurate, the prefix will contain the instance id.
Also, more of a nit but can we, please, keep the comments in this test with capital letter and full stop too? :D

Reply:
Right. Fixed.
Also I tried to add Capital letter and full stop. I hope I didn't miss any comment.